extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -347,22 +396,27 @@ private static void extractClientVersionAndKey() throws IOException, ExtractionE
 
         try {
             key = Parser.matchGroup1("INNERTUBE_API_KEY\":\"([0-9a-zA-Z_-]+?)\"", html);
-        } catch (Parser.RegexException e) {
+            keyAndVersionExtracted = true;
+        } catch (final Parser.RegexException e) {
             try {
                 key = Parser.matchGroup1("innertubeApiKey\":\"([0-9a-zA-Z_-]+?)\"", html);
-            } catch (Parser.RegexException ignored) {
+                keyAndVersionExtracted = true;
+            } catch (final Parser.RegexException ignored) {
             }
         }
+        keyAndVersionExtracted = false;
```

Comment:
This is also problematic. You should set it to false inside the last catch block, otherwise it will always be false. Also, I'd move the block of code throwing the parsing exception inside that same catch block, instead of letting that be done by the caller

Reply:
So, this is what I should do, with the removal of the throwing of `ParsingException`s in the two callers, or not?

```java
        try {
            key = Parser.matchGroup1("INNERTUBE_API_KEY\":\"([0-9a-zA-Z_-]+?)\"", html);
        } catch (final Parser.RegexException e1) {
            try {
                key = Parser.matchGroup1("innertubeApiKey\":\"([0-9a-zA-Z_-]+?)\"", html);
            } catch (final Parser.RegexException e2) {
                keyAndVersionExtracted = false;
                throw new ParsingException("Could not extract client version and key");
            }
        }
        keyAndVersionExtracted = true;
```

Reply:
Yes, looks good